validate.go

```diff
@@ -2,15 +2,265 @@ package main
 
 import (
 	"fmt"
-	"strings"
 
-	onelog "github.com/francoispqt/onelog"
+	appsv1 "github.com/kubewarden/k8s-objects/api/apps/v1"
+	batchv1 "github.com/kubewarden/k8s-objects/api/batch/v1"
 	corev1 "github.com/kubewarden/k8s-objects/api/core/v1"
+	apimachineryv1 "github.com/kubewarden/k8s-objects/apimachinery/pkg/apis/meta/v1"
 	kubewarden "github.com/kubewarden/policy-sdk-go"
+	kubernetes "github.com/kubewarden/policy-sdk-go/pkg/capabilities/kubernetes"
 	kubewarden_protocol "github.com/kubewarden/policy-sdk-go/protocol"
 	"github.com/mailru/easyjson"
 )
 
+const DEPLOYMENT_KIND = "deployment"
+const REPLICASET_KIND = "replicaset"
+const STATEFULSET_KIND = "statefulset"
+const DAEMONSET_KIND = "daemonset"
+const REPLICATIONCONTROLLER_KIND = "replicationController"
+const CRONJOB_KIND = "cronjob"
+const JOB_KIND = "job"
+const POD_KIND = "Pod"
+
+func getNamespace(validationRequest kubewarden_protocol.ValidationRequest) (*corev1.Namespace, error) {
+
+	metadata, err := getResourceMetadata(validationRequest)
+	if err != nil {
+		return nil, err
+	}
+	host := getWapcHost()
+
+	resourceRequest := kubernetes.GetResourceRequest{
+		APIVersion: "v1",
+		Kind:       "Namespace",
+		Name:       metadata.Namespace,
+	}
+
+	responseBytes, err := kubernetes.GetResource(&host, resourceRequest)
+	if err != nil {
+		return nil, fmt.Errorf("Cannot get namespace data: %s", err)
+	}
+	namespace := &corev1.Namespace{}
+	if err := easyjson.Unmarshal(responseBytes, namespace); err != nil {
+		return nil, fmt.Errorf("Cannot parse namespace data: %s", err)
+	}
+	return namespace, nil
+}
+
+func validateResourceLabels(namespaceLabels map[string]string, request kubewarden_protocol.ValidationRequest, settings Settings) ([]byte, error) {
+	labelsToPropagate := make(map[string]string)
+	for _, label := range settings.PropagatedLabels {
+		if value, namespace_has_label := namespaceLabels[label]; namespace_has_label {
+			labelsToPropagate[label] = value
+		}
+	}
+	return updateResourceLabels(request, labelsToPropagate)
+
+}
+func getResourceMetadata(object kubewarden_protocol.ValidationRequest) (*apimachineryv1.ObjectMeta, error) {
```

Comment:
This function is used only by the `getNamespace` function, the purpose is to extract the object metadata from the request, so that we can understand which namespace is owning the resource.

We can obtain this information in a faster way: the `AdmissionRequest` has a `Namespace` attribute that already holds this information. See [here](https://pkg.go.dev/github.com/kubewarden/policy-sdk-go@v0.4.1/protocol#KubernetesAdmissionRequest).

I propose to use this attribute and just remove this function entirely